mail_template_multi_report/models/email_template.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# © 2016 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+
+from openerp import api, fields, models
+from openerp.tools.safe_eval import safe_eval
+
+
+class EmailTemplate(models.Model):
+
+    _inherit = 'email.template'
+
+    report_line_ids = fields.One2many(
+        'email.template.report.line', 'template_id', string='Other Reports')
+
+    @api.model
+    def generate_email_batch(self, template_id, res_ids, fields=None):
+        results = super(EmailTemplate, self).generate_email_batch(
+            template_id, res_ids, fields=fields)
+
+        template = self.browse(template_id)
+
+        for report_line in template.report_line_ids:
+            records = self.env[template.model_id.model].browse(res_ids)
+
+            for rec in records:
+                condition = report_line.condition
+
+                if condition and condition.strip():
+                    condition_result = self.render_template(
+                        condition, template.model, rec.id)
+
+                    if not condition_result or not safe_eval(condition_result):
+                        continue
```

Comment:
Not sure if I understand this, but are you rendering the template before checking if you need it? I think it makes more sense to check first and render only if needed.